lib/maths/analytics/CBoostedTreeFactory.cc

```diff
@@ -119,6 +119,15 @@ auto validInputStream(core::CDataSearcher& restoreSearcher) {
     }
     return decltype(restoreSearcher.search(1, 1)){};
 }
+
+double minBoundary(const maths::analytics::CBoostedTreeHyperparameters::TDoubleParameter parameter,
+                   double maxBoundary,
+                   double interval) {
+    maxBoundary = parameter.toSearchValue(maxBoundary);
+    interval = parameter.toSearchValue(interval);
+    double minBoundary{maxBoundary - interval};
+    return parameter.fromSearchValue(minBoundary);
+}
```

Comment:
My inclination is to make this generic even if you only use for TDoubleParameter at present:
```suggestion
template<typename T>
double minBoundary(const CBoostedTreeParameter<T>& parameter,
                   T maxBoundary,
                   T interval) {
    maxBoundary = parameter.toSearchValue(maxBoundary);
    interval = parameter.toSearchValue(interval);
    T minBoundary{maxBoundary - interval};
    return parameter.fromSearchValue(minBoundary);
}
```